Update cloud cli docs with windows, new instructions

greg-mckeon opened the pull request:
Added the latest, feel free to change! Changes are still being added by product team so not ready to fully merge into mega branch yet

merging into #4191

[ ] needs docs review

greg-mckeon committed:
update cloud cli docs with windows, new instructions

mirnawong1 committed:
Merge branch 'mwong-cloud-cli-beta' into cloud-cli-2

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -1,107 +1,161 @@
+If the dbt help text appears, deactivate your current dbt Core install.
+
+2. Install the dbt Cloud CLI with Brew: 
 
```
do users need to install homebrew first? http://brew.sh/

greg-mckeon replied:
Yes, but I haven't seen any struggles with that step. I think we could add a link to brew?

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -1,107 +1,161 @@
+:::
+
+### Install and update with Brew - MacOS
+
```
given homebrew isn’t going to be supported for core, is this still right? do we need to say anything about [bundles](https://github.com/dbt-labs/dbt-core-bundles)?

greg-mckeon replied:
This is still right for Cloud CLI. We might change to recommending manual installs for all platforms, but for now it's right.

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -1,107 +1,161 @@
+1. Download the latest Windows release for your platform from [GitHub](https://github.com/dbt-labs/dbt-cli/releases).
 
-The following instructions are for setting up the dbt Cloud CLI. 
+2. Extract the `dbt-cloud-cli.exe` file into the same folder as your dbt project.
```
since there are many options, how do users know what release asset to download? for instance, if they have windows, do they download dbt_0.29.9_windows_amd64.tar.gz, or dbt_0.29.9_windows_arm64.tar.gz, or dbt_0.29.9_windows_armv6.tar.gz? when i downloaded those three, i didn't see 'dbt-cloud-cli.exe'. i saw 'dbt.exe'.

greg-mckeon replied:
ah, yes, this should be dbt.exe!

you pick the file based on your CPU architecture. The vast majority will want the amd64 version, as theyre likely running AMD/Intel CPUs.

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
````diff
@@ -1,107 +1,161 @@
-```
+### Manually install - Linux 
+
+1. Download the latest Linux release for your platform from [GitHub](https://github.com/dbt-labs/dbt-cli/releases).
````
same question as above

greg-mckeon replied:
Still CPU architecture based - Linux users should be relatively used to doing this.

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -1,107 +1,161 @@
+Have features you'd like to see in the Cloud CLI? Reach out to us on Slack!
 
 ## Glossary
 
```
is it replace `dbt` or replace `dbt-cloud-cli.exe`?

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -1,107 +1,161 @@
 
-3. Navigate to [your profile](https://cloud.getdbt.com/settings/profile) and enable the **Beta** flag under **Experimental Features.**
+Note that if you are using VS Code, you'll need to restart your IDE to pick up modified environment variables.
+:::
```
should we not use ide since we have a dbt cloud ide? should we instead say:

If you're using VS Code, remember to restart it so it can pick up modified environment variables.

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -1,107 +1,161 @@
+1. Download the latest Windows release for your platform from [GitHub](https://github.com/dbt-labs/dbt-cli/releases).
 
-The following instructions are for setting up the dbt Cloud CLI. 
+2. Extract the `dbt-cloud-cli.exe` file into the same folder as your dbt project.
```
do they need to run any commands to verify it's installed? 

Run the following command to verify the installation: dbt --help

If the help text doesn't show you're using the dbt Cloud CLI, ensure you've deactivated pyenv or venv and don't have a global dbt version installed

greg-mckeon replied:
yeah, we should add that here too.

greg-mckeon commented on website/docs/docs/cloud/cloud-cli-installation.md:
````diff
@@ -1,107 +1,161 @@
 ```
 
-You can find your project ID by selecting your project and clicking on **Develop** in the navigation bar. Your project ID is the number in the URL: https://cloud.getdbt.com/develop/26228/projects/PROJECT_ID.
+6. In your `dbt_project.yml` file, ensure there is a section titled "Cloud". This section is required to have a `project-id` field with a valid [project ID](#glossary). 
````
This is not implemented yet, so we're blocked from merging.

greg-mckeon replied:
This is now implemented - we should merge ASAP!

greg-mckeon committed:
address review feedback

Commit:
updates

Commit:
Merge branch 'cloud-cli-2' of https://github.com/dbt-labs/docs.getdbt.com into cloud-cli-2

mirnawong1 commented on website/docs/docs/cloud/configure-cloud-cli.md:
```diff
@@ -0,0 +1,65 @@
+
+#IS IT CLOUD OR DBT-CLOUD PER ISSUE https://github.com/dbt-labs/docs.getdbt.com/issues/4014
+
+cloud: 
```
@greg-mckeon is it dbt-cloud or cloud? referencing issue https://github.com/dbt-labs/docs.getdbt.com/issues/4014

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -1,110 +1,122 @@
+* Speedier, lower cost builds.
+* Significant platform improvements, to be released over the coming months.
+
+Note, the dbt Cloud CLI is a separate tool powered by [dbt Cloud](/docs/cloud/about-cloud/dbt-cloud-features). It differs from [dbt Core](https://github.com/dbt-labs/dbt-core), which provides dbt through its open-source command-line tool.
```
ADD INFO HOW MUST HAVE DBT CLOUD ACCOUNT TO USE (MAYBE NEED PREREQS)

Commit:
fold in Greg's feedback

mirnawong1 commented on website/docs/docs/cloud/configure-cloud-cli.md:
```diff
@@ -0,0 +1,65 @@
+## Using dbt Cloud CLI
+
+The dbt Cloud CLI is a drop-in replacement for dbt Core. When you invoke a dbt command, that command is sent to dbt Cloud for processing. Since this is still invoking dbt under the hood, you'll still need to run `dbt deps`, followed by the same model build commands you typically run.
+
```
@greg-mckeon are there any commands that can't work in the cloud cli? (e.g. dbt init?)

greg-mckeon replied:
Full list:

```sh
Available Commands:
  build         Run all seeds, models, snapshots, and tests in DAG order
  cancel        Cancel the most recent invocation
  clean         Delete all folders in the clean-targets list (usually the dbt_packages and target directories.)
  clone         Create clones of selected nodes based on their location in the manifest provided to --state.
  compile       Generates executable SQL from source, model, test and analysis files.
  deps          Pull the most recent version of the dependencies listed in packages.yml
  docs          Generate or serve the documentation website for your project
  help          Help about any command
  list          List the resources in your project
  parse         Parses the project and provides information on performance
  reattach      Reattach to the most recent invocation to retrieve logs and artifacts
  retry         Retry the nodes that failed in the previous run.
  run           Compile SQL and execute against the current target database.
  run-operation Run the named macro with any supplied arguments.
  seed          Load data from csv files into your data warehouse.
  show          Generates executable SQL for a named resource or inline query, runs that
SQL, and returns a preview of the results. Does not materialize anything to
the warehouse.
  snapshot      Execute snapshots defined in your project
  source        Manage your project's sources
  test          Runs tests on data in deployed models.
```

Commit:
update to dbt-cloud

mirnawong1 committed:
Merge branch 'mwong-cloud-cli-beta' into cloud-cli-2

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -1,110 +1,122 @@
+* Significant platform improvements, to be released over the coming months.
+
+Note, the dbt Cloud CLI is a separate tool powered by [dbt Cloud](/docs/cloud/about-cloud/dbt-cloud-features). It differs from [dbt Core](https://github.com/dbt-labs/dbt-core), which provides dbt through its open-source command-line tool.
 
```
greg to verify commands are the same (maybe a few not needed in cloud cli, like dbt init)

greg-mckeon replied:
Nearly all are the exact same. The only difference I know of, as of now, is that you need to quote selection syntax - https://dbtlabs.atlassian.net/browse/CLI-140

greg-mckeon replied:
Nearly all are the exact same. The only difference I know of, as of now, is that you need to quote selection syntax - https://dbtlabs.atlassian.net/browse/CLI-140

mirnawong1 replied:
gotcha! i've addressed this in the main pr: https://github.com/dbt-labs/docs.getdbt.com/pull/4173

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
````diff
@@ -1,110 +1,122 @@
-brew update
-brew upgrade dbt-cloud-cli
-```
+## Update dbt Cloud CLI
````
any manual update for linux?

greg-mckeon replied:
Same as windows - redownload via the link

Commit:
final tweaks

Commit:
update numbers

Commit:
add a next steps

Commit:
add card

mirnawong1 committed:
Merge branch 'mwong-cloud-cli-beta' into mwong-cloud-cli-links

Commit:
add overview page

Commit:
add cloud cli to relevant pages

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
````diff
@@ -1,110 +1,116 @@
-which dbt
-dbt --help
-```
+The dbt Cloud CLI, powered by [dbt Cloud](/docs/cloud/about-cloud/dbt-cloud-features), is distinct from [dbt Core](https://github.com/dbt-labs/dbt-core), an open-source command-line tool for dbt. Both tools are command lines that use the same dbt commands, but the dbt Cloud CLI is tailored to run specifically on dbt Cloud's infrastructure.
````
```suggestion
The dbt Cloud CLI and [dbt Core](https://github.com/dbt-labs/dbt-core), an open-source project, are both command line tools that let you run your dbt projects and use the same dbt commands. The key difference is that the dbt Cloud CLI is designed to work specifically with dbt Cloud's infrastructure.
```

mirnawong1 committed:
Update website/docs/docs/cloud/cloud-cli-installation.md

Commit:
cloud cli updates

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

Commit:
update command page

Commit:
clarify language

Commit:
tweak language

mirnawong1 committed:
Merge branch 'mwong-cloud-cli-beta' into cloud-cli-2

mirnawong1 commented on website/docs/docs/cloud/about-cloud-develop.md:
```diff
@@ -0,0 +1,19 @@
+dbt Cloud offers a fast and reliable way to work on your dbt project. You can use it through an integrated development environment (IDE) in your browser or a dbt Cloud-powered command line interface (CLI):
+
+- **[dbt Cloud CLI](/docs/cloud/cloud-cli-installation)** &mdash; Allows you to develop and run dbt commands from your local command line or code editor against your dbt Cloud development environment.
+  - Note, the dbt Cloud CLI is distinct from [dbt Core](https://github.com/dbt-labs/dbt-core), an open-source command-line tool for dbt. Both tools are command lines that use the same dbt commands, however, the dbt Cloud CLI is tailored to run specifically on dbt Cloud's infrastructure.
```
is this clear @greg-mckeon @jtcohen6 ?

```suggestion
  - The dbt Cloud CLI and [dbt Core](https://github.com/dbt-labs/dbt-core), an open-source project, are both command line tools that use the same dbt commands. The key difference is that the dbt Cloud CLI is designed to work specifically with dbt Cloud's infrastructure and integrates those builds with all [dbt Cloud features](/docs/cloud/about-cloud/dbt-cloud-features).
```

greg-mckeon replied:
I would either leave this off, or clarify the value:

`The dbt Cloud CLI and [dbt Core](https://github.com/dbt-labs/dbt-core), an open-source project, are both command line tools that run the same dbt commands. The dbt Cloud CLI  integrates with your dbt Cloud development environment, providing faster and cheaper builds, centralized access management and access to all dbt Cloud features.`

mirnawong1 replied:
ok gotcha! i'm just trying to make sure users don't get confused with the 'cli' term we use and associate with dbt core, and the dbt cloud cli.

Commit:
add linux

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -1,110 +1,116 @@
+* Secure credential storage in the dbt Cloud platform.
+* Automatic deferral of build artifacts to your Cloud project's production environment. 
+* Speedier, lower cost builds.
+* Significant platform improvements, to be released over the coming months.
```
```suggestion
* Support for dbt Mesh (cross-project `ref`),
* Significant platform improvements, to be released over the coming months.
```

mirnawong1 committed:
Update website/docs/docs/cloud/cloud-cli-installation.md

Commit:
update to pp

Commit:
Merge branch 'cloud-cli-2' of https://github.com/dbt-labs/docs.getdbt.com into cloud-cli-2

greg-mckeon commented on website/docs/docs/cloud/about-cloud-develop.md:
```diff
@@ -0,0 +1,19 @@
+- **[dbt Cloud CLI](/docs/cloud/cloud-cli-installation)** &mdash; Allows you to develop and run dbt commands from your local command line or code editor against your dbt Cloud development environment.
+  - Note, the dbt Cloud CLI is distinct from [dbt Core](https://github.com/dbt-labs/dbt-core), an open-source command-line tool for dbt. Both tools are command lines that use the same dbt commands, however, the dbt Cloud CLI is tailored to run specifically on dbt Cloud's infrastructure.
+
+- **[dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-in-the-cloud)** &mdash; Develop directly in your browser, making dbt project development efficient by compiling code into SQL and managing project changes seamlessly using an intuitive user interface (UI).
```
This is really about not requiring setup of dbt, right?

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

greg-mckeon reviewed (commented):
LGTM, few comments

Commit:
turn into tabs

dave-connors-3 committed:
add defer page to cloud develop docs

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop.md

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop.md

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

mirnawong1 committed:
Update about-cloud-develop.md

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
````diff
@@ -1,110 +1,185 @@
 ```bash
-brew update
-brew upgrade dbt-cloud-cli
+dbt deps
````
removing per @greg-mckeon 
```suggestion
```

mirnawong1 commented on website/docs/docs/cloud/configure-cloud-cli.md:
```diff
@@ -0,0 +1,63 @@
+
+## Use the dbt Cloud CLI
+
+The dbt Cloud CLI shares the same set of commands as dbt Core. When you invoke a dbt command, that command is sent to dbt Cloud for processing. Since this is still invoking dbt under the hood, you'll still need to run `dbt deps`, followed by the same model build commands you typically run.
```
```suggestion
The dbt Cloud CLI shares the same set of commands as dbt Core. When you invoke a dbt command, that command is sent to dbt Cloud for processing. 
```

mirnawong1 replied:
will add info about dbt commands

mirnawong1 replied:
add a note for user to run `dbt deps` if they change a package version or remove a package

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -1,110 +1,185 @@
+* Support for dbt Mesh (cross-project `ref`),
+* Significant platform improvements, to be released over the coming months.
+
+The dbt Cloud CLI and [dbt Core](https://github.com/dbt-labs/dbt-core), an open-source project, are both command line tools that let you run your dbt projects and use the same dbt commands. The key difference is that the dbt Cloud CLI is designed to work specifically with dbt Cloud's infrastructure.
```
```suggestion
The dbt Cloud CLI and [dbt Core](https://github.com/dbt-labs/dbt-core), an open-source project, are both command line tools that let you run your dbt projects and use the same dbt commands. The key difference is that the dbt Cloud CLI is designed to work specifically with dbt Cloud's infrastructure and integrates those builds with all [dbt Cloud features](/docs/cloud/about-cloud/dbt-cloud-features).
```

mirnawong1 committed:
Update website/docs/docs/cloud/cloud-cli-installation.md

Commit:
update commands

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

mirnawong1 committed:
Merge branch 'mwong-mega-sl-clou-cli' into cloud-cli-2

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

mirnawong1 committed:
Update website/docs/docs/running-a-dbt-project/run-your-dbt-projects.md

mirnawong1 committed:
Update website/docs/docs/core/about-the-cli.md

Commit:
add dbt mesh

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

Commit:
improve ux

greg-mckeon committed:
Update website/docs/docs/cloud/cloud-cli-installation.md

Co-authored-by: mirnawong1 <89008547+mirnawong1@users.noreply.github.com>

greg-mckeon committed:
Update website/docs/docs/cloud/configure-cloud-cli.md

Co-authored-by: mirnawong1 <89008547+mirnawong1@users.noreply.github.com>

greg-mckeon commented:
I think the last thing we need is to add a line about dbt deps not being required anymore. I'd love to ship this ASAP, since our current docs are now wrong after our most recent changes.

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -1,110 +1,184 @@
+1. Download the latest Windows release for your platform from [GitHub](https://github.com/dbt-labs/dbt-cli/releases).
 
-Follow the same process in [Installing dbt Cloud CLI](#manually-install-windows-only) and replace the existing `dbt` executable with the new one. You should not have to go through the security steps again.
+2. Extract the `dbt.exe` executeable into the same folder as your dbt project.
```
```suggestion
2. Extract the `dbt.exe` executable into the same folder as your dbt project.
```

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
````diff
@@ -1,110 +1,184 @@
-```bash
-cd ~/dbt-projects/jaffle_shop
-```
+Advanced users can configure multiple projects to use the same Cloud CLI executeable by adding it to their PATH environment variable in their shell profile.
````
```suggestion
Advanced users can configure multiple projects to use the same Cloud CLI executable by adding it to their PATH environment variable in their shell profile.
```

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

mirnawong1 committed:
Update cloud-cli-installation.md

mirnawong1 committed:
Update website/docs/docs/cloud/cloud-cli-installation.md

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

mirnawong1 committed:
Merge branch 'mwong-mega-sl-clou-cli' into cloud-cli-2

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

Commit:
update intro

mirnawong1 committed:
Merge branch 'mwong-mega-sl-clou-cli' into cloud-cli-2

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

mirnawong1 committed:
Merge branch 'mwong-cloud-cli-links' into dc-defer-in-cloud

mirnawong1 committed:
Update cloud-cli-installation.md

mirnawong1 committed:
Update website/docs/docs/cloud/cloud-cli-installation.md

mirnawong1 committed:
Update cloud-cli-installation.md

mirnawong1 committed:
Merge branch 'mwong-cloud-cli-links' into dc-defer-in-cloud

runleonarun commented on website/docs/docs/cloud/configure-cloud-cli.md:
```diff
@@ -0,0 +1,65 @@
+---
+title: Configure dbt Cloud CLI
+id: configure-cloud-cli
+description: "Instructions on how to configure the dbt Cloud CLI"
```
The "next" button currently links to the IDE docs because it's the next category. Do we want this behavior?

runleonarun commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -1,14 +1,13 @@
 ---
 
-:::warning Alpha functionality 
+:::info Public preview functionality
```
Should this be a reusable snippet so we can keep it in sync with the other one?

runleonarun commented on website/docs/docs/cloud/configure-cloud-cli.md:
```diff
@@ -0,0 +1,65 @@
+description: "Instructions on how to configure the dbt Cloud CLI"
+---
+
+:::info Public preview functionality
```
Same feedback: can this be a reusable snippet?

runleonarun commented on website/docs/docs/cloud/configure-cloud-cli.md:
````diff
@@ -0,0 +1,65 @@
+        export DBT_CLOUD_API_KEY="1234" # Replace 1234 with your API key   
+        ```
+
+   - In Powershell, add an environment variable: IS THIS MISSING SOMETHING?
````
I'm not a Windows user, but we can point them to these docs maybe 🤷🏻 
```suggestion
  - In Powershell, add an environment variable. Refer to [About environment variables](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_environment_variables?view=powershell-7.3) for more information.
```

runleonarun commented on website/docs/docs/cloud/configure-cloud-cli.md:
````diff
@@ -0,0 +1,65 @@
+cd ~/dbt-projects/jaffle_shop
+```
+
+4. In your `dbt_project.yml` file, ensure there is a section titled `dbt-cloud`. This section is required to have a `project-id` field with a valid project ID. 
````
Should they add this, or will it be there already?
```suggestion
4. In your `dbt_project.yml` file, add a section titled `dbt-cloud`. This section needs a `project-id` field with a valid project ID. 
```

runleonarun commented on website/docs/docs/cloud/configure-cloud-cli.md:
````diff
@@ -0,0 +1,65 @@
+    project-id: PROJECT_ID
+```
+
+- To find your project ID, go to **Develop** in the navigation menu. Select the dbt Cloud project URL, such as `https://cloud.getdbt.com/develop/26228/projects123456`, where the project ID is `123456`.
````
Suggest smoothing this flow a tiny bit. 
```suggestion
- To find your project ID, select **Develop** in the dbt Cloud navigation menu. You can use this URL to find the project ID. For example, in `https://cloud.getdbt.com/develop/26228/projects123456`, the project ID is `123456`.
```

runleonarun commented on website/docs/docs/cloud/configure-cloud-cli.md:
```diff
@@ -0,0 +1,65 @@
+
+## Use the dbt Cloud CLI
+
+The dbt Cloud CLI shares the same set of commands as dbt Core. When you invoke a dbt command, that command is sent to dbt Cloud for processing. 
```
Suggest making this sentence active.
```suggestion
The dbt Cloud CLI shares the same set of commands as dbt Core. dbt Cloud processes the dbt commands you invoke. 
```

runleonarun commented on website/docs/docs/cloud/configure-cloud-cli.md:
```diff
@@ -0,0 +1,65 @@
+
+The dbt Cloud CLI shares the same set of commands as dbt Core. When you invoke a dbt command, that command is sent to dbt Cloud for processing. 
+
+The dbt Cloud CLI supports [project dependencies](/docs/collaborate/govern/project-dependencies), which is an exciting way to depend on another project using the metadata service in dbt Cloud. It instantly resolves references (or  `ref`) to public models defined in other projects. You don't need to execute or analyze these upstream models yourself. Instead, you treat them as an API that returns a ...
```
I tried to clarify "it" by changing it to dbt Cloud CLI, but should it be "project dependencies?" 
```suggestion
The dbt Cloud CLI supports [project dependencies](/docs/collaborate/govern/project-dependencies), which enables you to depend on another project using the metadata service in dbt Cloud. Project dependencies instantly resolves references (or  `ref`) to public models defined in other projects. You don't need to execute or analyze these upstream models yourself. Instead, you treat them as an API that returns a dataset.
```

mirnawong1 replied:
ah yes it should be project dependencies. thanks for flagging this!

runleonarun commented on website/docs/docs/cloud/configure-cloud-cli.md:
```diff
@@ -0,0 +1,65 @@
+
+The dbt Cloud CLI supports [project dependencies](/docs/collaborate/govern/project-dependencies), which is an exciting way to depend on another project using the metadata service in dbt Cloud. It instantly resolves references (or  `ref`) to public models defined in other projects. You don't need to execute or analyze these upstream models yourself. Instead, you treat them as an API that returns a ...
+
+Share feedback or request features you'd like to see on the [dbt community Slack](https://getdbt.slack.com/archives/C05M77P54FL).
```
```suggestion
Share feedback or request features you'd like to see in the [dbt community Slack](https://getdbt.slack.com/archives/C05M77P54FL).
```

runleonarun commented on website/sidebars.js:
```diff
@@ -117,26 +117,40 @@ const sidebarSettings = {
-                "docs/cloud/dbt-cloud-ide/lint-format",
-                "docs/cloud/dbt-cloud-ide/dbt-cloud-tips",
+                "docs/cloud/about-cloud-develop",
+                {
```
Love the way this new TOC flows!

runleonarun commented on website/docs/docs/cloud/configure-cloud-cli.md:
````diff
@@ -0,0 +1,65 @@
+        ```bash
+        export DBT_CLOUD_API_KEY="1234" # Replace 1234 with your API key   
+        ```
+
````
Form line 34 down the text turns blue in GitHub. I think this means there's a space or code error before it. It looks fine in the output, so maybe nbd

mirnawong1 replied:
thank you Leona! looks like Greg made some updates

runleonarun commented on website/docs/docs/cloud/about-cloud-develop.md:
```diff
@@ -0,0 +1,20 @@
+pagination_next: "docs/cloud/cloud-cli-installation"
+---
+
+dbt Cloud offers a fast and reliable way to work on your dbt project. You can use it through an integrated development environment (IDE) in your browser or a dbt Cloud-powered command line interface (CLI):
```
suggest comparing the two modes browser or cli. Also suggest removing "it." 
```suggestion
dbt Cloud offers a fast and reliable way to work on your dbt project. You can develop in your browser using an integrated development environment (IDE) or in a dbt Cloud-powered command line interface (CLI):
```

runleonarun commented on website/docs/docs/cloud/about-cloud-develop.md:
```diff
@@ -0,0 +1,20 @@
+
+- **[dbt Cloud CLI](/docs/cloud/cloud-cli-installation)** &mdash; Allows you to develop and run dbt commands from your local command line or code editor against your dbt Cloud development environment.
+
+  - Something to note is that the dbt Cloud CLI and the open-sourced dbt Core are both command line tools that let you run your dbt projects and use the same dbt commands. The key difference is that the dbt Cloud CLI is designed to work specifically with dbt Cloud's infrastructure and integrates those builds with all [dbt Cloud features](/docs/cloud/about-cloud/dbt-cloud-features).
```
Suggest making this a note ☝🏻

runleonarun commented on website/docs/docs/cloud/about-cloud-develop.md:
```diff
@@ -0,0 +1,20 @@
+
+  - Something to note is that the dbt Cloud CLI and the open-sourced dbt Core are both command line tools that let you run your dbt projects and use the same dbt commands. The key difference is that the dbt Cloud CLI is designed to work specifically with dbt Cloud's infrastructure and integrates those builds with all [dbt Cloud features](/docs/cloud/about-cloud/dbt-cloud-features).
+
+- **[dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-in-the-cloud)** &mdash; Develop directly in your browser, making dbt project development efficient by compiling code into SQL and managing project changes seamlessly using an intuitive user interface (UI).
```
Suggest not identifying "UI" unless you're going to use it later on the page or in the paragraph.
```suggestion
- **[dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-in-the-cloud)** &mdash; Develop directly in your browser, making dbt project development efficient by compiling code into SQL and managing project changes seamlessly using an intuitive user interface.
```

runleonarun commented on website/docs/docs/cloud/about-cloud-develop.md:
```diff
@@ -0,0 +1,20 @@
+
+- **[dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-in-the-cloud)** &mdash; Develop directly in your browser, making dbt project development efficient by compiling code into SQL and managing project changes seamlessly using an intuitive user interface (UI).
+
+This documentation section provides detailed instructions on setting up the dbt Cloud CLI and dbt Cloud IDE. To get started with dbt development, you'll need a [developer](/docs/cloud/manage-access/seats-and-users) account. For a more comprehensive setup guide, refer to our [quickstart guides](/quickstarts).
```
```suggestion
The following sections provide detailed instructions on setting up the dbt Cloud CLI and dbt Cloud IDE. To get started with dbt development, you'll need a [developer](/docs/cloud/manage-access/seats-and-users) account. For a more comprehensive guide about developing in dbt, refer to our [quickstart guides](/quickstarts).
```

runleonarun reviewed (commented):
@mirnawong1 This looks great! I had a few suggestions. Let me know if you have any questions!

greg-mckeon committed:
add install feedback

greg-mckeon committed:
add credential file

mirnawong1 committed:
Update website/docs/docs/about-setup.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/docs/running-a-dbt-project/run-your-dbt-projects.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/docs/running-a-dbt-project/run-your-dbt-projects.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/docs/cloud/configure-cloud-cli.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

mirnawong1 committed:
Update about-cloud-develop.md

mirnawong1 committed:
Merge branch 'cloud-cli-2' into dc-defer-in-cloud

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -70,12 +105,11 @@ Note that if you are using VS Code, you'll need to restart it to pick up modifie
 </TabItem>
 
-<TabItem value="linux" label="Linux">
+<TabItem value="linux" label="Linux (native executeable)">
```
```suggestion
<TabItem value="linux" label="Linux (native executable)">
```

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -95,91 +129,46 @@ Advanced users can configure multiple projects to use the same Cloud CLI executa
 
 </TabItem>
-<TabItem value="windowslinux" label="Windows and Linux">
+<TabItem value="windowslinux" label="Windows and Linux (executeables)">
```
```suggestion
<TabItem value="windowslinux" label="Windows and Linux (executable)">
```

mirnawong1 committed:
Update website/docs/docs/cloud/configure-cloud-cli.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

mirnawong1 committed:
Update cloud-cli-installation.md

mirnawong1 committed:
Update website/docs/docs/cloud/cloud-cli-installation.md

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

mirnawong1 committed:
Update website/docs/docs/cloud/cloud-cli-installation.md

mirnawong1 committed:
Update website/docs/docs/cloud/configure-cloud-cli.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/docs/cloud/configure-cloud-cli.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

Commit:
updates

mirnawong1 committed:
Merge branch 'cloud-cli-2' into mwong-cloud-cli-links

mirnawong1 committed:
Merge branch 'cloud-cli-2' into dc-defer-in-cloud

Commit:
update pages and add redirect

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud/dbt-cloud-features.md

mirnawong1 committed:
Update develop-in-the-cloud.md

mirnawong1 committed:
add cloud cli links (#4179)

<p><a
href="https://www.notion.so/dbtlabs/Prod-Docs-Beta-Cloud-CLI-Oct-9th-a389341b7c1f45e2b7c162e4de591d90?d=2b3208f1ff8c49698fde7b3946b86e91">Docs
project</a></p>

[docs
project](https://www.notion.so/dbtlabs/Prod-Docs-Beta-Cloud-CLI-a389341b7c1f45e2b7c162e4de591d90)

this pr adds cloud cli links to pages/modifies cloud cli wording, and
defer docs. merging into #4192

[x] needs Technical review
[x] needs docs review

mirnawong1 committed:
Merge branch 'cloud-cli-2' into dc-defer-in-cloud

Commit:
update permissions

mirnawong1 committed:
Merge branch 'mwong-mega-sl-clou-cli' into cloud-cli-2

mirnawong1 committed:
Merge branch 'cloud-cli-2' into dc-defer-in-cloud

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop-defer.md

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop-defer.md

mirnawong1 committed:
Update about-cloud-develop-defer.md

mirnawong1 committed:
Update about-cloud-develop-defer.md

Commit:
rename cli to core

mirnawong1 committed:
Merge branch 'mwong-mega-sl-clou-cli' into cloud-cli-2

Commit:
fix broken links

Commit:
Merge branch 'cloud-cli-2' of https://github.com/dbt-labs/docs.getdbt.com into cloud-cli-2

mirnawong1 committed:
Merge branch 'cloud-cli-2' into dc-defer-in-cloud

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop-defer.md

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop-defer.md

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop-defer.md

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop-defer.md

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop-defer.md

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop-defer.md

mirnawong1 committed:
Update about-cloud-develop-defer.md

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop-defer.md

Commit:
add defer screenshot

Commit:
better pic

Commit:
add venv step

greg-mckeon commented on website/docs/docs/cloud/configure-cloud-cli.md:
```diff
@@ -0,0 +1,80 @@
+1. Ensure you meet the prerequisites above.
+
+2. Download your credentials from dbt Cloud by clicking on the **Try the dbt Cloud CLI** banner on the dbt Cloud homepage.
+3. Follow the banner instructions and download the config file to `~/.dbt/dbt_cloud.yml`. The config file looks like:
```
This is correct for Mac and Linux, but should be: C:\Users\yourusername\\.dbt\dbt_cloud.yml on windows

mirnawong1 committed:
Update configure-cloud-cli.md

add windows note

mirnawong1 committed:
Update website/docs/docs/cloud/about-cloud-develop-defer.md

Co-authored-by: dave-connors-3 <73915542+dave-connors-3@users.noreply.github.com>

mirnawong1 committed:
Merge branch 'cloud-cli-2' into dc-defer-in-cloud

mirnawong1 committed:
Update configure-cloud-cli.md

mirnawong1 committed:
Update cloud-cli-installation.md

mirnawong1 committed:
add defer page to cloud develop docs (#4229)

@dave-connors-3's pr which adds a doc explaining how def works in the
ide and cloud cli

## Checklist
[ ] needs Dave Connors review

mirnawong1 committed:
Update cloud-cli-installation.md

add space

Commit:
update

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
````diff
@@ -29,64 +27,120 @@ dbt commands are run against dbt Cloud's infrastructure and benefit from:
+  dbt-env\Scripts\activate            # activate the environment for Windows
+  ```
+
+3. Create an alias to activate your dbt environment with every new shell window or session. You can add the following to your shell's configuration file (for example, $HOME/.bashrc, $HOME/.zshrc) while replacing `<PATH_TO_VIRTUAL_ENV_CONFIG>` with the path to your virtual environment configuration:
````
step 3 if for mac/linux only -- what about windows?

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
````diff
@@ -29,64 +27,120 @@ dbt commands are run against dbt Cloud's infrastructure and benefit from:
-```
+  ```bash
+  brew tap dbt-labs/dbt-cli
+  brew install dbt
````
per benoit, this command should be same command as update one:

`brew install dbt` or `brew upgrade dbt-cloud-cli` -- is it dbt-cloud-cli or dbt?

greg-mckeon replied:
it is dbt. this one is correct

Commit:
add

Commit:
Merge branch 'cloud-cli-2' of https://github.com/dbt-labs/docs.getdbt.com into cloud-cli-2

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
````diff
@@ -29,64 +27,119 @@ dbt commands are run against dbt Cloud's infrastructure and benefit from:
+  dbt-env\Scripts\activate            # activate the environment for Windows
+  ```
+
+3. (Mac and Linux only) Create an alias to activate your dbt environment with every new shell window or session. You can add the following to your shell's configuration file (for example, $HOME/.bashrc, $HOME/.zshrc) while replacing `<PATH_TO_VIRTUAL_ENV_CONFIG>` with the path to your virtual environment configuration:
````
per benoit, this is for mac and linux - what is step for windows?

greg-mckeon replied:
We don't include an alias in our other docs for venv, because it's difficult to explain and many users won't have permissions to do so. I think we can leave it off here.

mirnawong1 committed:
Update cloud-cli-installation.md

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
````diff
@@ -95,90 +148,51 @@ Advanced users can configure multiple projects to use the same Cloud CLI executa
-```yaml
-# dbt_project.yml
-name:
+To update the dbt Cloud CLI, run `brew upgrade dbt-cloud-cli`. 
````
per benoit:

- this command should be same command as intall one: `brew install dbt` vs `brew upgrade dbt-cloud-cli ` -- is it `dbt` or `dbt-cloud-cli`?
- to update in brew, users can run `brew install` to install latest version. is it `brew upgrade` or `brew install`?

greg-mckeon replied:
either works, but I believe the standard is brew upgrade. it should also be `brew install dbt`, good catch

mirnawong1 committed:
Update cloud-cli-installation.md

mirnawong1 commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -95,90 +148,57 @@ Advanced users can configure multiple projects to use the same Cloud CLI executa
+<TabItem value="linux" label="Linux (executable)">
 
-- To find your project ID, go to **Develop** in the navigation menu. Select the dbt Cloud project URL, such as `https://cloud.getdbt.com/develop/26228/projects123456`, where the project ID is `123456`.
+To update, follow the same process explained in [Windows](/docs/cloud/cloud-cli-installation?install=linux#install-dbt-cloud-cli) and replace the existing `dbt` executable with the new one.
```
per Benoit, linux just uses `dbt` file

greg-mckeon replied:
yes, that is right. the file is named `dbt` instead of `dbt.exe`

mirnawong1 committed:
Merge branch 'mwong-mega-sl-clou-cli' into cloud-cli-2

greg-mckeon commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -29,64 +27,119 @@ dbt commands are run against dbt Cloud's infrastructure and benefit from:
+
+One of the benefits of the dbt Cloud CLI is that there is no need to manage Python environments. We provide this Python package for users who are using the dbt Cloud CLI in place of dbt Core for easy compatibility with existing environments.
+
+It replaces dbt Core, but this change can be avoided by using the native install method and configuring your PATH or by creating a new virtual environment. To switch back to dbt Core, follow the dbt Core installation instructions.
```
Could we delete the first two sentences, and just say "Installing the dbt Cloud CLI with pip replaces dbt Core. This change can be avoided by using the native install method and configuring your PATH or by creating a new virtual environment. 

Otherwise, to switch back to dbt Core, uninstall the Cloud CLI and follow the dbt Core installation instructions."

greg-mckeon commented on website/docs/docs/cloud/cloud-cli-installation.md:
```diff
@@ -29,64 +27,119 @@ dbt commands are run against dbt Cloud's infrastructure and benefit from:
+
+### Install a virtual environment
+
+We recommend using virtual environments (venv) to namespace pip modules.
```
"to namespace the cloud cli"

greg-mckeon commented on website/docs/docs/cloud/cloud-cli-installation.md:
````diff
@@ -29,64 +27,119 @@ dbt commands are run against dbt Cloud's infrastructure and benefit from:
+
+1. Create a new venv:
+   ```shell
+   python3 -m venv dbt-env
````
I would name this venv "dbt-cloud" so that there isn't confusion if they have followed the core install docs.

So: 

python3 -m venv dbt-cloud

greg-mckeon commented on website/docs/docs/cloud/cloud-cli-installation.md:
````diff
@@ -29,64 +27,119 @@ dbt commands are run against dbt Cloud's infrastructure and benefit from:
+
+2. Activate the virtual environment each time you create a shell window or session:
+  ```shell   
+  source dbt-env/bin/activate         # activate the environment for Mac and Linux OR
````
source dbt-cloud/bin/activate

greg-mckeon committed:
Update cloud-cli-installation.md

Pip updates

Commit:
add config files

Commit:
Merge branch 'cloud-cli-2' of https://github.com/dbt-labs/docs.getdbt.com into cloud-cli-2

Commit:
fix

mirnawong1 commented on website/docs/docs/cloud/configure-cloud-cli.md:
```diff
@@ -0,0 +1,91 @@
+
+1. Ensure you meet the prerequisites above.
+
+2. Download your credentials from dbt Cloud by clicking on the **Try the dbt Cloud CLI** banner on the dbt Cloud homepage. Or you can download the credentials from the following links depending on your region:
```
```suggestion
2. Download your credentials from dbt Cloud by clicking on the **Try the dbt Cloud CLI** banner on the dbt Cloud homepage. Alternatively, if you're in dbt Cloud, you can download the credentials from the links provided based on your region:
```

mirnawong1 committed:
Update website/docs/docs/cloud/configure-cloud-cli.md

mirnawong1 committed:
Update configure-cloud-cli.md

mirnawong1 committed:
Update cloud-cli-installation.md

mirnawong1 committed:
Update cloud-cli-installation.md

mirnawong1 committed:
Merge branch 'mwong-mega-sl-clou-cli' into cloud-cli-2

Commit:
update quickstarts

Commit:
add tenant info

mirnawong1 committed:
Merge branch 'mwong-mega-sl-clou-cli' into cloud-cli-2
